lib/mpl_toolkits/mplot3d/tests/test_axes3d.py

```diff
@@ -1774,21 +1832,17 @@ def test_rotate():
         ax.view_init(0, 0, roll)
         ax.figure.canvas.draw()
 
-        # drag mouse horizontally to change azimuth
-        dx = 0.1
-        dy = 0.2
+        # drag mouse horizontally to change orientation
         ax._button_press(
             mock_event(ax, button=MouseButton.LEFT, xdata=0, ydata=0))
         ax._on_move(
             mock_event(ax, button=MouseButton.LEFT,
-                           xdata=dx*ax._pseudo_w, ydata=dy*ax._pseudo_h))
+                           xdata=0.5*ax._pseudo_w, ydata=0*ax._pseudo_h))
         ax.figure.canvas.draw()
-        roll_radians = np.deg2rad(ax.roll)
-        cs = np.cos(roll_radians)
-        sn = np.sin(roll_radians)
-        assert ax.elev == (-dy*180*cs + dx*180*sn)
-        assert ax.azim == (-dy*180*sn - dx*180*cs)
-        assert ax.roll == roll
+
+        assert np.isclose(ax.elev, roll)
+        assert np.isclose(ax.azim, -90)
+        assert np.isclose(ax.roll, 0)
```

Comment:
> These numbers seem too "clean" for a mouse movement?

Well, yes, but these angles are easy enough to understand what the result should be. For 'more arbitrary' numbers, it is hard to know the result, short of running the calculation. The test is more of a 'freezing any errors in place' than a validation.


> And why does elev equal roll and roll 0?

Because dragging the mouse rotated the figure!